src/Stack/Package.hs

```diff
@@ -1062,6 +1064,7 @@ getCabalFileName
     => Path Abs Dir -- ^ package directory
     -> m (Path Abs File)
 getCabalFileName pkgDir = do
+    liftIO $ hpack pkgDir
```

Comment:
I think this is the right place to call hpack, but this changes the meaning of `getCabalFileName` enough to warrant renaming it to `findOrGenerateCabalFile`.  The new name isn't perfect (makes it sound like it'll always generate a cabal file if it's missing), however, this is better because it lets the reader know it may perform file system mutations (not merely a "get").